airbyte-oauth/src/main/java/io/airbyte/oauth/flows/FacebookMarketingOAuthFlow.java

```diff
@@ -0,0 +1,107 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.oauth.flows;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.oauth.BaseOAuthFlow;
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Following docs from
+ * https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow
+ */
+public class FacebookMarketingOAuthFlow extends BaseOAuthFlow {
+
+  private static final String CONSENT_URL = "https://www.facebook.com/v11.0/dialog/oauth";
+  private static final String ACCESS_TOKEN_URL = "https://graph.facebook.com/v11.0/oauth/access_token";
+
+  public FacebookMarketingOAuthFlow(ConfigRepository configRepository) {
+    super(configRepository);
+  }
+
+  @VisibleForTesting
+  FacebookMarketingOAuthFlow(ConfigRepository configRepository, HttpClient httpClient) {
+    super(configRepository, httpClient);
+  }
+
+  @Override
+  protected String getBaseConsentUrl() {
+    return CONSENT_URL;
+  }
+
+  @Override
+  protected Map<String, String> getConsentQueryParameters(UUID definitionId, String clientId, String redirectUrl) {
+    return ImmutableMap.<String, String>builder()
+        // required
+        .put("client_id", clientId)
+        .put("redirect_uri", redirectUrl)
+        .put("state", getState(definitionId))
+        // optional
+        .put("response_type", "code")
+        .put("scope", "ads_management,ads_read,read_insights")
+        .build();
+  }
+
+  @Override
+  protected String extractCodeParameter(Map<String, Object> queryParams) throws IOException {
+    if (queryParams.containsKey("code")) {
+      return (String) queryParams.get("code");
+    } else {
+      throw new IOException("Undefined 'code' from consent redirected url.");
+    }
+  }
+
+  @Override
+  protected String getAccessTokenUrl() {
+    return ACCESS_TOKEN_URL;
+  }
+
+  @Override
+  protected Map<String, String> getAccessTokenQueryParameters(String clientId, String clientSecret, String authCode, String redirectUrl) {
+    return ImmutableMap.<String, String>builder()
+        // required
+        .put("client_id", clientId)
+        .put("redirect_uri", redirectUrl)
+        .put("client_secret", clientSecret)
+        .put("code", authCode)
+        .build();
+  }
+
+  @Override
+  protected Map<String, Object> extractRefreshToken(JsonNode data) throws IOException {
+    if (data.has("access_token")) {
```

Comment:
can you add a comment explaining why `extractRefreshToken` is returning what seems to be an access Token? 😅

(i know why but the next person reading may not)